lib/LANraragi/Utils/Plugins.pm

```diff
@@ -163,10 +163,8 @@ sub use_plugin {
         # Execute the plugin, appending the custom args at the end
         if ( $pluginfo{type} eq "script" ) {
             eval { %plugin_result = LANraragi::Model::Plugins::exec_script_plugin( $plugin, $input, @settings ); };
-        }
-
-        if ( $pluginfo{type} eq "metadata" ) {
-            eval { %plugin_result = LANraragi::Model::Plugins::exec_metadata_plugin( $plugin, $id, $input, @settings ); };
+        } elsif ( $pluginfo{type} eq "metadata" ) {
+            %plugin_result = LANraragi::Model::Plugins::exec_metadata_plugin( $plugin, $id, $input, @settings );
         }
 
         if ($@) {
```

Comment:
You're probably already aware, but this check would be removed once script plugins have an internal try/catch as well

Reply:
Yes, I wanted to treat it in the same way as the metadata plugins.